lib/linter.js

```diff
@@ -312,34 +313,48 @@ export default class Linter {
     }
 
     let hasBOM = options.source.codePointAt(0) === 0xfe_ff;
-    let source = hasBOM ? options.source.slice(1) : options.source;
+    let rawSource = hasBOM ? options.source.slice(1) : options.source;
 
-    let templateAST;
+    let templateInfos = extractTemplates(rawSource, options.filePath);
 
-    try {
```

Comment:
this is mostly the same, it just got moved into the loop that iterates over all the detected templates in the file